smarts/core/provider.py

```diff
@@ -37,6 +37,7 @@ def merge(self, other: "ProviderState"):
         assert our_vehicles.isdisjoint(other_vehicles)
 
         self.vehicles += other.vehicles
+        self.dt = max(self.dt, other.dt, key=lambda x: x if x else 0)
```

Comment:
This is the key bug fix here.  When this was missing, line 591 of smarts.py did not function as intended.

Reply:
Yes.  (Because either argument may be None, so the code got ugly using `if`-statements.)